enhancements/sig-architecture/138-workload-completion/README.md

```diff
@@ -0,0 +1,363 @@
+    JsonPaths []JsonPathsCompletion `json:"jsonPaths,omitempty"`
+
+    // JsonPathsOperator defines the logical operator used to join the result of multiple paths.
+    // If left empty, this will default to "And"
+    // +optional
+    JsonPathsOperator JsonPathsOperator `json:"jsonPathsOperator,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=WellKnownCompletions;JSONPaths
+type CompletionType string
+
+const (
+    // WellKnownCompletionType represents standard completions for some common types, which
+    // is reflected with a hardcoded rule only for types in k8s.io/api
+    WellKnownCompletionsType CompletionType = "WellKnownCompletions"
+
+    // JsonPathsCompletionType represents that a manifest will be considered completed when the given json paths
+    // match specified values.
+    JsonPathsCompletionType CompletionType = "JSONPaths"
+)
+
+// +kubebuilder:validation:Enum=And;Or
+type JsonPathsOperator string
+
+const (
+  // JsonPathsLogicalAnd represents that the result of the JsonPaths rule will be the AND of all paths
+  JsonPathsLogicalAnd JsonPathsOperator = "And"
+  // JsonPathsLogicalOr represents that the result of the JsonPaths rule will be the OR of all paths
+  JsonPathsLogicalOr JsonPathsOperator = "Or"
+)
+
+type JsonPathsCompletion struct {
+    // Path represents the json path of the field to be checked for equality.
+    // The path must point to a field with single value in the type of integer, bool or string.
+    // If the path points to a non-existing field, the manifest will not be considered complete.
+    // If the path points to a structure, map or slice, the manifest will not be considered complete.
+    // Ref to https://kubernetes.io/docs/reference/kubectl/jsonpath/ on how to write a jsonPath.
+    // +kubebuilder:validation:Required
+    // +required
+    Path string `json:"path"`
+
+    // Value represents the expected value at the given path for the manifest to be considered complete.
+    // The value will be parsed as an integer, bool, or string depending on the type of the field retrieved from path.
+    // If parsing to the appropriate type fails, the manifest will not be considered complete.
+    // When the field and parsed value are strictly equivalent, the rule will be considered completed.
+    // +kubebuilder:validation:Required
+    // +required
+    Value string `json:"value"`
+}
+```
+
+To enable automated garbage collection after completion, `ttlSecondsAfterFinished` will be added to `deleteOption`
+
+```go
+type DeleteOption struct {
+  ...
+
+  // TTLSecondsAfterFinished limits the lifetime of a ManifestWork that has satisfied
+  // all completionRules configured for its manifests. If this field is set, and
+  // the manifestwork has completed, then it is elligible to be automatically deleted.
+  // If this field is unset, the manifestwork won't be automatically deleted even afer completion.
+  // If this field is set to zero, the manfiestwork becomes elligible to be deleted immediately
```

Comment:
Will the ManifestWork itself be deleted, or will the K8s resources specified in its manifests be deleted from the managed cluster instead?

Reply:
The ManifestWork itself will be deleted, just like a Job with ttlSecondsAfterFinished.

The k8s resources specified in its manifests will respect the configured deletion options as before. So by default they will be deleted as well.